File: 1350-remove-sub-folders-from-the-filesystem/remove-sub-folders-from-the-filesystem.cpp
class Solution {
public:


    vector<string> removeSubfolders(vector<string>& folder) {
        vector<string>ans;
        sort(folder.begin(),folder.end());
        ans.push_back(folder[0]);
        for(int i=1;i<folder.size();i++){
            string checker = ans.back();
            int n = checker.size();
            if(folder[i].substr(0,n)!=checker || folder[i].size()>n && folder[i][n]!='/'){
                ans.push_back(folder[i]);
            }
        }
        return ans;

    }
};

File: 1350-remove-sub-folders-from-the-filesystem/remove-sub-folders-from-the-filesystem.go
func removeSubfolders(folder []string) []string {
    ans:= []string{}
    sort.Strings(folder);
    ans = append(ans,folder[0])
    for _,val:=range folder[1:] {
        checker:=ans[len(ans)-1]
        if strings.HasPrefix(val,checker) && len(val)>len(checker) && val[len(checker)]=='/'{
            continue
        }else{
            ans = append(ans,val);
        }
    }
    return ans;

}